zbuf/batch.go

```diff
@@ -72,3 +72,32 @@ func CopyPuller(w Writer, p Puller) error {
 		b.Unref()
 	}
 }
+
+func PullerReader(p Puller) Reader {
+	return &pullerReader{p: p}
+}
+
+type pullerReader struct {
+	p     Puller
+	batch Batch
+	idx   int
+}
+
+var _ Reader = (*pullerReader)(nil)
+
+func (r *pullerReader) Read() (*zng.Record, error) {
+	if r.batch == nil {
+		batch, err := r.p.Pull()
+		if err != nil || batch == nil {
+			return nil, err
+		}
+		r.batch = batch
+		r.idx = 0
+	}
```

Comment:
Although we try to avoid zero-length batches, they do sometimes occur, so you might want to handle them.